File: Algorithms/InsertionSort/InsertionSort.java
class InsertionSort{

    public static void main(String[] args){
        int[] arrayToSort = {4,3,4,56,6,8,7,100};
        sort(arrayToSort);
        for(int i =0; i< arrayToSort.length;i++){
            System.out.println(arrayToSort[i]);
        }

    }

    public static void sort(int[] arrayToSort){
        int pivot;
        for(int i =0; i< arrayToSort.length;i++){
            pivot = arrayToSort[i];
            int j = i - 1;
            while(j >=0 && arrayToSort[j] > pivot){
              arrayToSort[j+1] = arrayToSort[j];
              j--;
            }
            arrayToSort[j+1] = pivot;
        }
    }
}

File: Algorithms/InsertionSort/InsertionSort.c
#include <stdio.h>

void sort(int *toSort, size_t arrayLength){
  int pivot = 0;
  for(int i=0; i<arrayLength;i++){
    pivot = toSort[i];
    int j = i -1;

    while(j >=0  && toSort[j] > pivot){
      toSort[j+1] = toSort[j];
      j--;
    }
    toSort[j+1] = pivot;


  }
}

int main(){
  int toSort[] = {4,3,4,56,6,8,7};
  size_t length = 7;
  sort(toSort, length);
  for(int i = 0; i < length; i++ ){
    printf("%d\n", toSort[i]);
  }
}
